torch/csrc/jit/codegen/cuda/kernel_ir.h

```diff
@@ -0,0 +1,271 @@
+
+#pragma once
+
+#include <torch/csrc/WindowsTorchApiMacro.h>
+#include <torch/csrc/jit/codegen/cuda/ir_base_nodes.h>
+
+// TODO: remove these
+#include <torch/csrc/jit/codegen/cuda/ir_interface_nodes.h>
+#include <torch/csrc/jit/codegen/cuda/ir_internal_nodes.h>
+
+#include <c10/util/Optional.h>
+
+#include <string>
+#include <unordered_map>
+#include <vector>
+
+namespace torch {
+namespace jit {
+namespace fuser {
+namespace kir {
+
+// TODO: Fill out TensorIndex, which is a list of Ints used to directly index a
+// TensorView. It is not the flattened index, which needs to be computed using
+// stride information.
+class TORCH_CUDA_API TensorIndex : public Val {
```

Comment:
Do we need `TORCH_CUDA_API` for this and the other classes in the `kir` namespace?

Reply:
Yes. Unless we have a good reason to explicitly hide them, the default for all external definitions is to use TORCH_CUDA_API.